internal/app/management/run.go

```diff
@@ -0,0 +1,456 @@
+							VolumeMounts: []apiv1.VolumeMount{
+								{
+									Name:      "module-config",
+									MountPath: configMapFilePath,
+								},
+							},
+						},
+					},
+					Volumes: []apiv1.Volume{
+						{
+							Name: "module-config",
+							VolumeSource: apiv1.VolumeSource{
+								ConfigMap: &apiv1.ConfigMapVolumeSource{
+									LocalObjectReference: apiv1.LocalObjectReference{
+										Name: moduleConfigMapName,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	deploymentsClient := k.client.AppsV1().Deployments(k.namespace)
+	_, err = deploymentsClient.Create(deployment)
+	if err != nil {
+		return nil, fmt.Errorf("error creating dispatcher deployment %+v", err)
+	}
+
+	var createResponse = &module.ModuleCreateResponse{
+		Name: id,
+	}
+
+	return createResponse, nil
+}
+
+func (s *server) Delete(ctx context.Context, r *module.ModuleDeleteRequest) (*module.ModuleDeleteResponse, error) {
+	// Find deployments with matching label and delete them
+	deploymentsClient := k.client.AppsV1().Deployments(k.namespace)
+	deployments, err := deploymentsClient.List(metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("id=%s", r.Name),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error listing deployments with name %s", r.Name)
+	}
+	for _, deployment := range deployments.Items {
+		if err := deploymentsClient.Delete(deployment.Name, nil); err != nil {
+			return nil, fmt.Errorf("error deleting deployment %s", deployment.Name)
+		}
+	}
+
+	// Find configmaps with matching label and delete them
+	configMapClient := k.client.CoreV1().ConfigMaps(k.namespace)
+	configmaps, err := configMapClient.List(metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("id=%s", r.Name),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error listing configmaps with name %s", r.Name)
+	}
+	for _, configmap := range configmaps.Items {
+		if err := configMapClient.Delete(configmap.Name, nil); err != nil {
+			return nil, fmt.Errorf("error deleting configmap %s", configmap.Name)
+		}
+	}
+
+	// Find the secrets with the matching label and delete them
+	secretsClient := k.client.CoreV1().Secrets(k.namespace)
+	secrets, err := secretsClient.List(metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("id=%s", r.Name),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error listing configmaps with name %s", r.Name)
+	}
+	for _, secret := range secrets.Items {
+		if err := secretsClient.Delete(secret.Name, nil); err != nil {
+			return nil, fmt.Errorf("error deleting secret %s", secret.Name)
+		}
+	}
+
+	var deleteResponse = &module.ModuleDeleteResponse{
+		Name: r.Name,
+	}
+
+	return deleteResponse, nil
+}
+
+func (s *server) List(ctx context.Context, r *module.ModuleListRequest) (*module.ModuleListResponse, error) {
+	// Find deployments with matching label
+	deploymentsClient := k.client.AppsV1().Deployments(k.namespace)
+	deployments, err := deploymentsClient.List(metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("createdBy=%s", mgmt.ID),
```

Comment:
This is really nice, we should look at adding similar filtering into the dispatcher when it polls jobs as it would likely remove unnecessary load by only returning jobs in the state we're interested in.